src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java

```diff
@@ -416,35 +425,34 @@ private PaymentTransactionInfoPlugin executeFollowUpTransaction(final Transactio
         final DateTime utcNow = clock.getUTCNow();
 
         final PaymentModificationResponse response;
-        try {
-            response = transactionExecutor.execute(transactionAmount, paymentProvider, pspReference, splitSettlementData);
-        } catch (final ModificationFailedException e) {
-            return new AdyenPaymentTransactionInfoPlugin(kbPaymentId, kbTransactionId, transactionType, amount, currency, PaymentServiceProviderResult.ERROR, utcNow, e);
+        response = transactionExecutor.execute(transactionAmount, paymentProvider, pspReference, splitSettlementData);
+        if (!response.isSuccess()) {
+            return new AdyenPaymentTransactionInfoPlugin(kbPaymentId, kbTransactionId, transactionType, amount, currency, Optional.<PaymentServiceProviderResult>absent(), utcNow, response);
```

Comment:
Ruby plugins have two tables: responses (for all calls going to the gateway) and transactions (summary of successful calls). A successful payment would have an entry in both tables. A failed payment only in responses.

We could either do something similar or introduce a new `success` column.